internal/controller/config/agent_config.go

```diff
@@ -0,0 +1,137 @@
+package config
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	agentcore "github.com/buildkite/agent/v3/core"
+)
+
+// AgentConfig stores shared parameters for things that run buildkite-agent in
+// one form or another. They should correspond to the flags for
+// `buildkite-agent start`. Note that not all agent flags make sense as config
+// options for agents running within a pod.
+type AgentConfig struct {
+	// Applies to agents within the k8s controller and within the pod
+	Endpoint *string `json:"endpoint,omitempty"`
+	NoHTTP2  bool    `json:"no-http2,omitempty"`
+
+	// Only applies to agents within the pod
+	Experiments               []string `json:"experiment,omitempty"`
+	Shell                     *string  `json:"shell,omitempty"`
+	NoColor                   bool     `json:"no-color,omitempty"`
+	StrictSingleHooks         bool     `json:"strict-single-hooks,omitempty"`
+	NoMultipartArtifactUpload bool     `json:"no-multipart-artifact-upload,omitempty"`
+	TraceContextEncoding      *string  `json:"trace-context-encoding,omitempty"`
+
+	// Applies differently depending on the container (agent server vs checkout/
+	// command container).
+	NoPTY            bool `json:"no-pty,omitempty"`
+	NoCommandEval    bool `json:"no-command-eval,omitempty"`
+	NoLocalHooks     bool `json:"no-local-hooks,omitempty"`
+	NoPlugins        bool `json:"no-plugins,omitempty"`
+	PluginValidation bool `json:"plugin-validation,omitempty"`
+
```

Comment:
Would it be good to have the signing warn option? 

This way if you mix agents you can at least avoid 🔥

Reply:
I added most of the signing and verification options, and made it default to `warn` if no key is specified. (If a key is specified, they could want `block` by default). WDYT?

Reply:
👍🏻 sounds great, this layered default is a bit more complex but also safer for those who don't even know what signing is which is a big plus.